### DISCUSSION TITLE
Popup Search Window

### DISCUSSION THREAD

**Discussion by ferdy-roz** [zed-industries/zed#23934] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Maybe as a configuration option the ability to turn the search everywhere (Cmd+Shift+F) command bring up a popup window over current buffer(s) like the JetBrains IDEs. Helps to not move focus away from the current buffer/tab.

Also related:

* The ability to see search results as you are typing
* Navigate search results w/ keyboard and be able to go to a search hit by hitting Enter

Apologies if there is an open issue for this (I tried searching) or, even more embarrassing, if this is already doable!

**Comment by cole-miller** [comment]
From the Zed team: thanks for the idea! This seems like something we should talk over in discussions.

**Reply by jv-np** [reply]
any progress? I would consider switch from vscode just for that, it's damn annoying needing to change my context just to search something.

**Comment by jonatino** [comment]
If I could upvote this a trillion times I would. Having codebase wide search quickly popup without making you lose your current code tab is way more intuitive imo. The lack of this feature is the only thing keeping me on jetbrains ide's otherwise I would've switched to vscode or zed long ago.

**Comment by bes** [comment]
And moving searches to their own docked area, with tabs for each search (like RustRover) would be _amazing_.

**Comment by ozacod** [comment]
You might want to check this out #46478

**Reply by jonatino** [reply]
That is PERFECT! Well done
